components/admin_console/permission_schemes_settings/edit_post_time_limit_modal/edit_post_time_limit_modal.tsx

```diff
@@ -34,21 +34,21 @@ export default function EditPostTimeLimitModal(props: Props) {
 
     const [saving, setSaving] = useState(false);
     const [errorMessage, setErrorMessage] = useState('');
-    const [postEditTimeLimit, setPostEditTimeLimit] = useState<number>(ServiceSettings?.PostEditTimeLimit || -1);
-    const [alwaysAllowPostEditing, setAlwaysAllowPostEditing] = useState(postEditTimeLimit > 0);
+    const [postEditTimeLimit, setPostEditTimeLimit] = useState<number>(ServiceSettings?.PostEditTimeLimit || Constants.UNSET_POST_EDIT_TIME_LIMIT);
+    const [alwaysAllowPostEditing, setAlwaysAllowPostEditing] = useState(postEditTimeLimit < 0);
 
     const save = async () => {
         setSaving(true);
         setErrorMessage('');
 
-        if (postEditTimeLimit.toString() === 'NaN' || postEditTimeLimit < -1 || postEditTimeLimit > INT32_MAX) {
+        if (postEditTimeLimit.toString() === 'NaN' || postEditTimeLimit < 0 || postEditTimeLimit > INT32_MAX) {
```

Comment:
we can safely remove or add a type check replacement for the `NaN` check, since with the new state this will never be anything else than of type `number`

```suggestion
        if (postEditTimeLimit < 0 || postEditTimeLimit > INT32_MAX) {
```

or to be ultra-secure

```suggestion
        if (typeof postEditTimeLimit !== 'number' || postEditTimeLimit < 0 || postEditTimeLimit > INT32_MAX) {
```

Reply:
I would keep 'NaN' check to eliminate the user keeps the input empty and save. When I try both of your suggestions, I see that user can leave input empty and save. Please let me know if I am missing sth here

Reply:
that is something that is unaccounted for ... so to handle this specific case we can add a different guard (the `toString()` method is to complicated)

```suggestion
        if (isNaN(postEditTimeLimit) || postEditTimeLimit < 0 || postEditTimeLimit > INT32_MAX) {
```

`isNaN` is a built in method to check for `NaN` specifically

Reply:
Good to know that. The change is done